src/schema/submissions.ts

```diff
@@ -160,15 +160,21 @@ export const resolvers: IResolvers<unknown, Context> = traceResolvers({
         };
       }
 
-      const existingPost = await getPostByUrl(cleanUrl, ctx, info);
+      const existingPost = await ctx
+        .getRepository(ArticlePost)
+        .findOneBy([{ url }, { canonicalUrl: url }]);
       if (existingPost) {
         if (existingPost.deleted) {
           return {
             result: 'rejected',
             reason: SubmissionFailErrorMessage.POST_DELETED,
           };
         }
-        return { result: 'exists', post: existingPost };
+
+        return {
+          result: 'exists',
+          post: await getPostByUrl(cleanUrl, ctx, info),
```

Comment:
This feels a bit duplicate from the one above.
Let's maybe see if it can be uniformed one way?

Reply:
This function is using graphorm, the change introduced above was using typeorm.